js/layout-javascript/news-feed-code.js

```diff
@@ -2337,99 +2380,126 @@ DynamicList.prototype.sendComment = function(id, value) {
     });
   }
 
-  _this.appendTempComment(id, value, guid, userFromDataSource);
+  var options = {
+    instance: _this,
+    config: _this.data,
+    id: _this.data.id,
+    uuid: _this.data.uuid,
+    container: _this.$container,
+    record: record,
+    comment: value,
+    commentGuid: guid
+  };
+
+  return Fliplet.Hooks.run('flListDataBeforeNewComment', options).then(function () {
+    value = options.comment;
+    guid = options.commentGuid;
 
-  var record = _.find(_this.listItems, { id: id });
+    if (!value) {
+      return Promise.resolve();
+    }
 
-  if (!record) {
-    return Promise.resolve();
-  }
+    _this.appendTempComment(id, value, guid, userFromDataSource);
 
-  if (_.get(record, 'commentCount')) {
-    record.commentCount++;
-  }
+    if (_.get(record, 'commentCount')) {
+      record.commentCount++;
+    }
 
-  _this.updateCommentCounter({
-    id: id,
-    record: record
-  });
+    _this.updateCommentCounter({
+      id: id,
+      record: record
+    });
 
-  userName = _.compact(_.map(_this.data.userNameFields, function (name) {
-    return _this.myUserData.isSaml2
-      ? _.get(userFromDataSource, 'data.' + name)
-      : _this.myUserData[name];
-  })).join(' ').trim();
+    userName = _.compact(_.map(_this.data.userNameFields, function (name) {
+      return _this.myUserData.isSaml2
+        ? _.get(userFromDataSource, 'data.' + name)
+        : _this.myUserData[name];
+    })).join(' ').trim();
 
-  var comment = {
-    fromName: userName,
-    user: _this.myUserData.isSaml2 ? userFromDataSource.data : _this.myUserData
-  };
+    var comment = {
+      fromName: userName,
+      user: _this.myUserData.isSaml2 ? userFromDataSource.data : _this.myUserData
+    };
 
-  _.assignIn(comment, { contentDataSourceEntryId: id });
+    _.assignIn(comment, { contentDataSourceEntryId: id });
 
-  var query;
+    var timestamp = (new Date()).toISOString();
 
-  var timestamp = (new Date()).toISOString();
+    // Get mentioned user(s)
+    var mentionRegexp = /\B@[a-z0-9_-]+/ig;
```

Comment:
what if a username contains a space, e.g. a fullname?

Reply:
`Fliplet.Utils.Users.getUsersToMention()` removes the spaces from existing users so only @mentions without spaces are considered.